packages/istanbul-lib-coverage/test/file.test.js

```diff
@@ -153,36 +153,36 @@ describe('base coverage', () => {
         const template = new FileCoverage({
             path: '/path/to/file',
             statementMap: {
-                1: loc(1, 1, 1, 100),
-                2: loc(2, 1, 2, 50),
-                3: loc(2, 51, 2, 100),
-                4: loc(2, 101, 3, 100)
+                0: loc(1, 1, 1, 100),
```

Comment:
It seems a bit odd to me that this new logic would have switched tests to being 0 offset vs., 1 offset.

Is this expected?

Reply:
@bcoe 

The previous incarnation of merge [preserved the object keys while zipping up the hit counters](https://github.com/istanbuljs/istanbuljs/blob/f6bb0b4f3e7596a4e2dcaa1d0695afe1f97e6eee/packages/istanbul-lib-coverage/lib/file-coverage.js#L169). [As those keys are just ascending integers](https://gist.github.com/JustinChristensen/aa256e0187dc638f7db08a1391136861#file-first-coverage-json-L626), and not a kind of identifier, it didn't seem particularly important to preserve them. Unless that behavior is documented somewhere, I suspect that the test was written with that unimportant implementation detail in mind, and not to validate an API's guarantees.

Reply:
Actually, I probably should check the spec on that.

Reply:
https://tc39.es/ecma262/multipage/ordinary-and-exotic-objects-behaviours.html#sec-ordinaryownpropertykeys

Yup.

Reply:
sounds good 👍